caas/kubernetes/provider/k8s_test.go

```diff
@@ -1185,7 +1217,8 @@ func (s *K8sBrokerSuite) TestEnsureCustomResourceDefinitionCreate(c *gc.C) {
 	ctrl := s.setupController(c)
 	defer ctrl.Finish()
 
-	podSpec := basicPodspec
+	podSpec := &caas.PodSpec{}
```

Comment:
I think we should change the test objects like `basicPodSpec` to be functions that return by-value or return new objects by ptr.
IMO this is just code smell.
```
podSpec := &caas.PodSpec{}
*podSpec = *basicPodspec
```

Reply:
sure changed